### ISSUE TITLE
Sometimes the last line between a clone pair is not the same.

### ISSUE THREAD

**Issue by cchaniotaki** [dspinellis/mpcd#1]
You may see some examples here: [C](https://htmlpreview.github.io/?https://github.com/Christina96/boilerplate-sample-matches/blob/main/C/redis-MDEwOlJlcG9zaXRvcnkxMDgxMTA3MDY=-flat-hyperloglog.c-redis-MDEwOlJlcG9zaXRvcnkxMDgxMTA3MDY=-flat-sds_35.c-21.html), [C++](https://htmlpreview.github.io/?https://github.com/Christina96/boilerplate-sample-matches/blob/main/Cpp/tesseract-MDEwOlJlcG9zaXRvcnkyMjg4NzA5NA==-flat-normstrngs.cpp-tesseract-MDEwOlJlcG9zaXRvcnkyMjg4NzA5NA==-flat-par_control.cpp-34.html), [C#](https://htmlpreview.github.io/?https://github.com/Christina96/boilerplate-sample-matches/blob/main/Csharp/npgsql-MDEwOlJlcG9zaXRvcnkyODgzNTc0-flat-TransactionTests.cs-npgsql-MDEwOlJlcG9zaXRvcnkyODgzNTc0-flat-TransactionTests.cs-0.html), [Python](https://htmlpreview.github.io/?https://github.com/Christina96/boilerplate-sample-matches/blob/main/Python/yolact-MDEwOlJlcG9zaXRvcnkxMzg3OTY2OTk=-flat-dcn_v2.py-yolact-MDEwOlJlcG9zaXRvcnkxMzg3OTY2OTk=-flat-dcn_v2.py-13.html), [Java](https://htmlpreview.github.io/?https://github.com/Christina96/boilerplate-sample-matches/blob/main/Java/motan-MDEwOlJlcG9zaXRvcnk1NjY3OTUyMQ==-flat-FailbackRegistry.java-motan-MDEwOlJlcG9zaXRvcnk1NjY3OTUyMQ==-flat-FailbackRegistry.java-5.html).

Additionally, I ran an example only with the two files in C.
On results.json you may see the following group:
```
{
    "tokens": 18,
    "groups": [
      {
        "start": 394,
        "end": 395,
        "filepath": "./sds_35.c"
      },
      {
        "start": 410,
        "end": 411,
        "filepath": "./sds_35.c"
      },
      {
        "start": 427,
        "end": 428,
        "filepath": "./sds_35.c"
      },
      {
        "start": 505,
        "end": 506,
        "filepath": "./hyperloglog.c"
      },
      {
        "start": 509,
        "end": 510,
        "filepath": "./hyperloglog.c"
      }
    ]
  }
```
The following pair on the last line is not the same:
```
 [{ "start": 505, "end": 506, "filepath": "./hyperloglog.c" },
   { "start": 410, "end": 411, "filepath": "./sds_35.c" }]
```
<img width="900" alt="example" src="https://github.com/dspinellis/mpcd/assets/15112370/907cf59b-9412-4c9e-a10d-fd31311d2ff7">

You can find all the example here: 
[example.zip](https://github.com/dspinellis/mpcd/files/11990424/example.zip)

**Comment by dspinellis** [comment]
Thank you for preparing this!  I cannot replicate the example with the current version (1.1.2). It generates a `results.json` file smaller (1912 lines) than the one you supplied. Please ensure you are running the latest version. This is (now) available through the `-V` option.

**Comment by dspinellis** [comment]
I was able to replicate it on another computer.  The lines are different at the level of tokens.
```
<hyperloglog.c tokenizer -l C -o line -c -f -s  | cat -n | more
   505  ID PLUS_EQUAL ( int ) ID ; ID & ID ; * ID ( int ) * /
   506  ID PLUS_PLUS ;

<sds_35.c tokenizer -l C -o line -c -f -s  | cat -n
  410  ID PLUS_EQUAL ( int ) ID ; ID & ID ; * ID ( int ) * /
   411  }
```

Note that both files contain spurious `*/` sequences. Are they corrupt or is this how they appear online? In any case, this shouldn't cause the observed issue.

**Comment by dspinellis** [comment]
The results depend on the order the files are processed. The following replicates the issue deterministically.
```sh
{ echo sds_35.c ; echo hyperloglog.c ; } | tokenizer -l C -o line -i - -c -f | mpcd -v -n 15 -j
```

**Comment by dspinellis** [comment]
Fixed in v1.1.3.
